Resources/Computing/ARCComputingElement.py

```diff
@@ -91,27 +108,38 @@ def submitJob( self, executableFile, proxy, numberOfJobs = 1 ):
     batchIDList = []
     stampDict = {}
 
+    # Do not need to loop over these ... the pilot proxy is the same for everyone
+    usercfg = arc.UserConfig()
+    usercfg.CredentialString(proxy)
+    endpoint = arc.Endpoint(cfgPath("gsiftp://", self.ceHost, ":2811/jobs"), arc.Endpoint.JOBSUBMIT,
+                            "org.nordugrid.gridftpjob")
+
     i = 0
     while i < numberOfJobs:
       i += 1
+      # The basic job description
+      job = arc.Job()
+      jobdescs = arc.JobDescriptionList()
+      # Get the job into the ARC way
       xrslName, diracStamp = self.__writeXRSL( executableFile )
-      cmd = ['arcsub', '-j', self.ceParameters['JobListFile'],
-             '-c', '%s' % self.ceHost, '%s' % xrslName ]
-      result = executeGridCommand( self.proxy, cmd, self.gridEnv )
+      if not arc.JobDescription_Parse(xrslName, jobdescs):
+        logger.msg(arc.ERROR, "Invalid job description")
+        sys.exit(1) 
```

Comment:
Do you really want to exit abruptly like this ? If so, I think (but I don't know why :-) ) you should use the DIRAC exit